feature_engine/encoding/mean_encoding.py

```diff
@@ -143,8 +156,20 @@ def fit(self, X: pd.DataFrame, y: pd.Series):
 
         self.encoder_dict_ = {}
 
+        y_mean = y.mean()
+        if self.a == 'auto':
+            y_var = y.var(ddof=0)
         for var in self.variables_:
-            self.encoder_dict_[var] = y.groupby(X[var]).mean().to_dict()
+            if self.a == 'auto':
+                damping = y.groupby(X[var]).var(ddof=0) / y_var
```

Comment:
Is this described in the article of micci barrera?

Reply:
Yes, 3.1 last paragraphs